cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -1337,37 +1342,31 @@ private boolean isNullOrDefault(@Nullable String namespace) {
   }
 
   /**
-   * Updates the token store to be used for the hive job, based upon the Explore container's credentials.
-   * This is because twill doesn't update the container_tokens on upon token refresh.
-   * See: https://issues.apache.org/jira/browse/TWILL-170
+   * Writes the {@link Credentials} of the current user to a file in the {@link #credentialsDir}, for a particular query
+   *
+   * @param queryHandle the query handler for the current operation
+   * @return a File where the credentials were written.
    */
-  private void updateTokenStore() throws IOException, ExploreException {
-    String hadoopTokenFileLocation = System.getenv(UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION);
-    if (hadoopTokenFileLocation == null) {
-      LOG.warn("Skipping update of token store due to failure to find environment variable '{}'.",
-               UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION);
-      return;
-    }
+  private File writeCredentialsFile(QueryHandle queryHandle) throws IOException, ExploreException {
+    File credentialsFile = new File(credentialsDir, queryHandle.getHandle());
```

Comment:
`writeCredentialsFile` is called from a method which generates the QueryHandle on the fly (so its unique per call to write credentials).